tests/integration_test.go

```diff
@@ -0,0 +1,368 @@
+
+	// Test against signature generated using ethers.js
+	require.Equal(t, sigHex, "f86108053294464646464646464646464646464646464646464646840406080a1ba06eaf33b72493638546d6294b39694c5e420fa55de74af6f1c87739da2582e231a016568d498cebfaa7ed80db99b7be76c31f5b78285cfa1794be3141dfae838f16")
+}
+
+func TestLedgerSignTyped1(t *testing.T) {
+	wallet, account := initWallet(t, gethaccounts.DefaultBaseDerivationPath)
+	defer wallet.Close()
+
+	messageStandard := map[string]interface{}{
+		"from": map[string]interface{}{
+			"name":   "Cow",
+			"wallet": "0xCD2a3d9F938E13CD947Ec05AbC7FE734Df8DD826",
+		},
+		"to": map[string]interface{}{
+			"name":   "Bob",
+			"wallet": "0xbBbBBBBbbBBBbbbBbbBbbbbBBbBbbbbBbBbbBBbB",
+		},
+		"contents": "Hello, Bob!",
+	}
+
+	typedData := createTypedDataPayload(messageStandard)
+
+	sigBytes, err := wallet.SignTypedData(account, typedData)
+	require.NoError(t, err)
+
+	sigHex := hex.EncodeToString(sigBytes)
+
+	require.Equal(t, sigHex, "fb35835539608d309ee5ee4b3dfbbc8cb4b591d7e8c9c745473848cbe1e13b037278226e2d6962b3b19145147314d9872ff853437e3ebd654d44aace09128acd1c")
+}
+
+func TestLedgerSignTyped2(t *testing.T) {
+	wallet, account := initWallet(t, gethaccounts.DefaultBaseDerivationPath)
+	defer wallet.Close()
+
+	messageStandard := map[string]interface{}{
+		"from": map[string]interface{}{
+			"name":   "Charlie",
+			"wallet": "0x1CfC9d8357cBE15E08Bb7084073B7E4ef790B625",
+		},
+		"to": map[string]interface{}{
+			"name":   "Delta",
+			"wallet": "0x53Fe71EDEFdF942dDE10834ed4d443A6df391F64",
+		},
+		"contents": "Message from Charlie to Delta!",
+	}
+
+	typedData := createTypedDataPayload(messageStandard)
+
+	sigBytes, err := wallet.SignTypedData(account, typedData)
+	require.NoError(t, err)
+
+	sigHex := hex.EncodeToString(sigBytes)
+
+	require.Equal(t, sigHex, "d929a56d69a98f3e491828fbd1555e66ddde17c8928a69704e710a9c34db1ab80314ffccf7014be6c8f819ca9c9603d59aad58cddaa1e6f43c7f66a6b9183c681c")
+}
+
+// Test signing TypedData from a non-default derivation path
+func TestLedgerSignTyped3(t *testing.T) {
+	path, err := gethaccounts.ParseDerivationPath("m/44'/60'/0'/0/1")
+	if err != nil {
+		panic("Could not parse derivation path")
+	}
+
+	wallet, account := initWallet(t, path)
+	defer wallet.Close()
+
+	messageStandard := map[string]interface{}{
+		"from": map[string]interface{}{
+			"name":   "Charlie",
+			"wallet": "0x1CfC9d8357cBE15E08Bb7084073B7E4ef790B625",
+		},
+		"to": map[string]interface{}{
+			"name":   "Delta",
+			"wallet": "0x53Fe71EDEFdF942dDE10834ed4d443A6df391F64",
+		},
+		"contents": "Message from Charlie to Delta!",
+	}
+
+	typedData := createTypedDataPayload(messageStandard)
+
+	sigBytes, err := wallet.SignTypedData(account, typedData)
+	require.NoError(t, err)
+
+	sigHex := hex.EncodeToString(sigBytes)
+
+	require.Equal(t, sigHex, "76984ce659f841975bdab7762ed9cb3c936791d1dcded3c0554147fca7accfdc543313669dcda04350990884e9e10c382fb20b722c123409a97c42ef6df617ca1c")
+}
+
+func CreateTx(
```

Comment:
Nit: no need to export this function
```suggestion
func createTx(
```